src/main/java/uk/gov/companieshouse/api/accounts/validation/RptTransactionValidator.java

```diff
@@ -24,14 +26,20 @@ public Errors validateRptTransaction(RptTransaction rptTransaction, Transaction
 
         boolean isMultipleYearFiler = getIsMultipleYearFiler(transaction);
 
-        if(!isMultipleYearFiler && rptTransaction.getBreakdown().getBalanceAtPeriodStart() != null ) {
+        if (!isMultipleYearFiler && rptTransaction.getBreakdown().getBalanceAtPeriodStart() != null) {
             addError(errors, unexpectedData, TRANSACTION_BREAKDOWN_PATH_BALANCE_AT_PERIOD_START);
             return errors;
         } else if (isMultipleYearFiler && rptTransaction.getBreakdown().getBalanceAtPeriodStart() == null) {
             addError(errors, mandatoryElementMissing, TRANSACTION_BREAKDOWN_PATH_BALANCE_AT_PERIOD_START);
             return errors;
         }
 
+        if (!((rptTransaction.getTransactionType().trim().equalsIgnoreCase(TRANSACTION_TYPE_MESSAGE[0]) ||
```

Comment:
As previous comment, if a Map used then can use the .containsValue method or with an Enum, the EnumUtils.isValidEnum method if we're only storing key in DB

Reply:
Which EnumUtils is that? I cannot find it.
Anyway, I agree with the idea of mapping to an enum, given a choice.
I don't think that maps are more efficient than a short list for very small collections.
Although something like Google's immutable map might be more efficient than many, you still end up calculating a hash for the string to look it up.

Reply:
"I still vividly remember the day that Wirth decided to replace the elegant data structure used in the compiler’s symbol table handler by a mundane linear list. In the original compiler, the objects in the symbol table had been sorted in a tree data structure (in identifier lexical order) for fast access, with a separate linear list representing their declaration order. One day Wirth decided that there really weren’t enough objects in a typical scope to make the sorted tree cost-effective. All of us Ph.D. students were horrified: it had taken time to implement the sorted tree, the solution was elegant, and it worked well – so why would one want to throw it away and replace it by something simpler, and even worse, something as prosaic as a linear list? But of course, Wirth was right, and the simplified compiler was both smaller and faster than its predecessor."
http://www.ics.uci.edu/~franz/Site/pubs-pdf/BC03.pdf